app/controllers/api/v1/events_by_month_controller.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    class EventsByMonthController < Api::V1::ApiController
+      def index
+        date = params[:date] ? Time.zone.parse(params[:date]) : Time.zone.now
+        @events = Event.on_month(date, current_api_v1_user)
+        @calendar = events_to_calendar_view(@events)
+        raise ActiveRecord::RecordNotFound, 'not found' unless @events
+      end
+
+      private
+
+      def events_to_calendar_view(events)
+        calendar = Hash.new { |hsh, key| hsh[key] = [] }
+        events.map { |event| calendar[event.start_time.strftime('%Y-%m-%d')].push(event_to_array(event)) }
```

Comment:
What would happen if we have more than one event for the same day?

Reply:
It push the event to the array. This is an example of the response with two events for the same day: `{
    "2021-09-09": [
        {
            "id": 2,
            "name": "Ad at tenetur sint.",
            "address": "98310 Heather Harbor",
            "start_time": "2021-09-09T00:00:00.000Z",
            "end_time": "2021-09-09T03:00:00.000Z",
            "updated_event_at": null,
            "cancelled": false,
            "updated_at": "2020-11-04T10:38:26.828Z"
        },
        {
            "id": 8,
            "name": "Qui ut saepe non.",
            "address": "7965 Heller Cliff",
            "start_time": "2021-09-09T00:00:00.000Z",
            "end_time": "2020-12-21T05:00:00.000Z",
            "updated_event_at": null,
            "cancelled": false,
            "updated_at": "2020-11-04T12:20:03.928Z"
        }
    ],...
}`